src/Automation/Scanner.cs

```diff
@@ -72,8 +74,25 @@ private IReadOnlyCollection<ScanResults> ExecuteScan(string scanId)
             return ExecutionWrapper.ExecuteCommand(() =>
             {
                 _scanTools.OutputFileHelper.SetScanId(scanId);
+                return SnapshotCommand.ExecuteScanAll(_config, _scanTools);
+            });
+        }
 
-                return SnapshotCommand.Execute(_config, _scanTools);
+        public Task<ScanResults> ScanAsync(string scanId, CancellationToken cancellationToken)
+        {
+            return ExecutionWrapper.ExecuteCommand(() =>
+            {
+                _scanTools.OutputFileHelper.SetScanId(scanId);
+                return SnapshotCommand.ExecuteScanAsync(_config, _scanTools, cancellationToken);
```

Comment:
> Concretely, can we remove ExecuteScanAsync and instead use return SnapshotCommand.ExecuteScanAllAsync(_config, _scanTools, cancellationToken).First();?

I think this was added for symmetry with the sync API.

Reply:
> There's a mismatch in the types. Task<ScanResults> and Task<IReadOnlyCollection<ScanResults>> are incompatible.

Could we transform the type with `ContinueWith`? Something like:

```
ScanAllAsync(scanId, cancellationToken).ContinueWith((task) => task.Result.FirstOrDefault(), TaskScheduler.Default);
```

Alternatively, returning a `ScanResultsCollection` instead of the existing return types sounds reasonable to me.

Reply:
Looks like this point is now moot given the other changes to the API, resolving this thread :)